src/app_service/service/message/service.nim

```diff
@@ -167,12 +168,21 @@ QtObject:
   
   proc handlePinnedMessagesUpdate(self: Service, pinnedMessages: seq[PinnedMessageUpdateDto]) =
     for pm in pinnedMessages:
-      let data = MessagePinUnpinArgs(chatId: pm.chatId, messageId: pm.messageId, actionInitiatedBy: pm.pinnedBy)
+      var chatId: string = ""
+      if (self.numOfPinnedMessagesPerChat.contains(pm.localChatId)):
+        # In 1-1 chats, the message's chatId is the localChatId
+        chatId = pm.localChatId
+      elif (self.numOfPinnedMessagesPerChat.contains(pm.chatId)):
+        chatId = pm.chatId
+
+      let data = MessagePinUnpinArgs(chatId: chatId, messageId: pm.messageId, actionInitiatedBy: pm.pinnedBy)
       if(pm.pinned):
-        self.numOfPinnedMessagesPerChat[pm.chatId] = self.getNumOfPinnedMessages(pm.chatId) + 1
+        if (chatId != "" and pm.pinnedBy != singletonInstance.userProfile.getPubKey()):
+          self.numOfPinnedMessagesPerChat[chatId] = self.numOfPinnedMessagesPerChat[chatId] + 1
```

Comment:
@jrainville I deliberately added `getNumOfPinnedMessages`, cause using it, we are safe to get the number of pinned messages for a chat regardless `chatId` is added as a key to the table or not (only because of this check there `if(self.numOfPinnedMessagesPerChat.hasKey(chatId)):`). So would good (safer) to call `getNumOfPinnedMessages` instead of direct access. The same for the line 185.

Reply:
Agreed. You added it after I started my own PR, and yours got merged first. That's why I have a conflict. I'll fix the conflict and use your function